File: Pattern44.py

#  ***
# *   *
# *   *
# *   *
#  ***
# *   *
# *   *
# *   *
#  ***


n = int(input("Enter the number:"))

for i in range(1, (n*2)+1 ):
    if i == 1 or i == n or i == n*2: #First, last and middle lines
        print(" " + "*"*(n-2))
    else:
        print("*"+" "*(n-2)+"*")

File: PATTERNS/Pattern44/Pattern44.go
package main

import "fmt"

func main() {
	var n int
	fmt.Print("Enter the row number:")
	fmt.Scanf("%d", &n)
	for i := 1; i <= 2*n-1; i++ {
		if i == 1 || i == n || i == 2*n-1 {
			fmt.Printf(" ")
		}
		for j := 1; j <= n; j++ {
			if i == 1 || i == n || i == 2*n-1 {
				if j > 1 && j < n {
					fmt.Printf("*")
				}
			} else {
				if j == 1 || j == n {
					fmt.Printf("*")
				} else {
					fmt.Printf(" ")
				}
			}
		}
		fmt.Printf("\n")
	}
}
